src/EnergyPlus/PluginManager.cc

```diff
@@ -55,41 +61,355 @@
+    int (*EP_PyRun_SimpleString)(const char *) = nullptr;
+    int (*EP_Py_FinalizeEx)() = nullptr;
+    void (*EP_PyErr_Fetch)(PyObjectWrap **, PyObjectWrap **, PyObjectWrap **) = nullptr;
+    PyObjectWrap (*EP_PyObject_Repr)(PyObjectWrap) = nullptr;
+    PyObjectWrap (*EP_PyUnicode_AsEncodedString)(PyObjectWrap, const char *, const char *) = nullptr;
+    char *(*EP_PyBytes_AsString)(PyObjectWrap) = nullptr;
+    PyObjectWrap (*EP_PyUnicode_DecodeFSDefault)(const char *) = nullptr;
+    PyObjectWrap (*EP_PyImport_Import)(PyObjectWrap) = nullptr;
+    void (*EP_Py_DECREF)(PyObjectWrap) = nullptr;
+    PyObjectWrap (*EP_PyErr_Occurred)() = nullptr;
+    PyObjectWrap (*EP_PyModule_GetDict)(PyObjectWrap) = nullptr;
+    PyObjectWrap (*EP_PyDict_GetItemString)(PyObjectWrap, const char *) = nullptr;
+    const char *(*EP_PyUnicode_AsUTF8)(PyObjectWrap) = nullptr;
+    PyObjectWrap (*EP_PyUnicode_AsUTF8String)(PyObjectWrap) = nullptr;
+    int (*EP_PyCallable_Check)(PyObjectWrap) = nullptr;
+    PyObjectWrap (*EP_PyObject_CallObject)(PyObjectWrap, PyObjectWrap) = nullptr;
+    PyObjectWrap (*EP_PyObject_GetAttrString)(PyObjectWrap, const char *) = nullptr;
+    PyObjectWrap (*EP_PyObject_CallFunction)(PyObjectWrap, const char *) = nullptr;
+    PyObjectWrap (*EP_PyObject_CallMethod)(PyObjectWrap, const char *, const char *) = nullptr;
+    int (*EP_PyList_Check)(PyObjectWrap) = nullptr;
+    unsigned long (*EP_PyList_Size)(PyObjectWrap) = nullptr;
+    PyObjectWrap (*EP_PyList_GetItem)(PyObjectWrap, size_t) = nullptr;
+    int (*EP_PyUnicode_Check)(PyObjectWrap) = nullptr;
+    int (*EP_PyLong_Check)(PyObjectWrap) = nullptr;
+    long (*EP_PyLong_AsLong)(PyObjectWrap) = nullptr;
+    void (*EP_Py_SetPythonHome)(const wchar_t *) = nullptr;
 
-    void registerNewCallback(int iCalledFrom, std::function<void ()> f)
+    void checkWrapperDLLFunction(const char *err, std::string const &pyFuncName, bool &errFlag)
+    {
+        if (err) {
+            std::string c = "Problem processing function \"" + pyFuncName + "\" in Python wrapper shared library";
+            ShowSevereError(c);
+            ShowContinueError(err);
+            errFlag = true;
+        }
+    }
+
+    void loadWrapperDLL()
+    {
+#if LINK_WITH_PYTHON == 1
+        // There are two ways to call E+: as an executable and as a library
+        // When calling E+ as a library, the plugin system is not available
+        // So when calling for plugins, the user must be calling as energyplus.exe
+        // In this case, the current executable path will be the E+ install root
+        // Which is where we find the Python Wrapper Shared Library
+        if (wrapperDLLHandle) return; // already found
+        std::string programPath = FileSystem::getProgramPath();
+        std::string programDir = FileSystem::getParentDirectoryPath(programPath);
+        std::string sanitizedProgramDir = PluginManager::sanitizedPath(programDir);
+        // set the path to the python library
+#ifdef _WIN32
```

Comment:
Don't need to protect _WIN64 ?

Reply:
@rraustad good eye, but _WIN32 is just a broad preprocessor definition that applies to multiple Windows platforms.  From the [definition](https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=vs-2019): 

> _WIN32 Defined as 1 when the compilation target is 32-bit ARM, 64-bit ARM, x86, or x64. Otherwise, undefined.

So it will be defined for our x86 or x64 builds.